### PR TITLE
Use date time column filters #382

### LINKED ISSUE #382: Change column filters for dates to use the date time colum filter in MRT
https://www.material-react-table.com/docs/guides/column-filtering#filter-variants

- add modified times to the directory dialog tables

- The delivered date and warranty should using date range column filter (Items Tables)

- all the other dates should use the datetime range column filter (creation and modification times )

- the date format for the creation and modification time changed (use .toLocaleString() method )

### PR DESCRIPTION
**Pull request by MatteoGuarnaccia5**
## Description

- Added datetime filters to creation and modified columns
- Added modified time column to directory dialogs
- Changed to use `formatDateTimeStrings` for `warranty_end_date` and `delivered_date`
- Added date filters for warranty and delivered columns

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking

closes #382

### CODE REVIEW

**Comment by joshuadkitenge** [comment]
can you use the formatDatetime funciton again. That format looks better

File: src/utils.tsx [hunk 1]
```diff
@@ -78,6 +78,10 @@ export const trimStringValues = (object: any): any => {
 
 export const formatDateTimeStrings = (dateTime: string): string => {
   const date = parseISO(dateTime);
-  const formattedDate = format(date, 'dd MMM yyyy HH:mm');
+  const formattedDate =
```
Comment (joshuadkitenge) [inline src/utils.tsx:81]: Can you change this function to have two variables `(date: String , includesTime (or withTime):boolean)`. and use the includesTime (or withTime) variable for the formatting.

If you look at the snapshots some of the created times does not include the time with the current implementation

Comment (joshuadkitenge) [reply src/utils.tsx:81]: image

File: src/manufacturer/manufacturer.component.tsx [hunk 2]
```diff
@@ -63,16 +63,16 @@ function ManufacturerComponent() {
...
+        accessorFn: (row) => new Date(row.modified_time),
         id: 'modified_time',
-        size: 250,
+        filterVariant: 'datetime-range',
+        size: 350,
         enableGrouping: false,
         Cell: ({ row }) =>
           row.original.modified_time &&
-          formatDateTimeStrings(row.original.modified_time),
+          formatDateTimeStrings(row.original.modified_time, false),
```
Comment (joshuadkitenge) [inline src/manufacturer/manufacturer.component.tsx:75]:
```suggestion
          formatDateTimeStrings(row.original.modified_time, true),
```

**Review by joshuadkitenge — CHANGES_REQUESTED** [review]
Looks good, you missed one in the manufacturers page. Can you also move the created time so it is appears to the right of the last modified time (still keep it hidden).

File: src/catalogue/items/CatalogueItemsDetailsPanel.component.tsx [hunk 3]
```diff
@@ -160,17 +160,27 @@ function CatalogueItemsDetailsPanel(props: CatalogueItemsDetailsPanelProps) {
               </Grid>
 
               <Grid item xs={12} sm={6} key={9}>
-                <Typography color="text.primary">Last Modified</Typography>
-                <Typography color="text.secondary">
-                  {formatDateTimeStrings(catalogueItemIdData.modified_time)}
-                </Typography>
+                <Grid item xs={12}>
```
Comment (joelvdavies) [inline src/catalogue/items/CatalogueItemsDetailsPanel.component.tsx:163]: I dont think this grid is necessary
```suggestion
```

File: src/catalogue/items/CatalogueItemsDetailsPanel.component.tsx [hunk 4]
```diff
@@ -160,17 +160,27 @@ function CatalogueItemsDetailsPanel(props: CatalogueItemsDetailsPanelProps) {
...
+                  </Typography>
+                </Grid>
               </Grid>
 
               <Grid item xs={12} sm={6} key={10}>
-                <Typography color="text.primary">Created</Typography>
-                <Typography color="text.secondary">
-                  {formatDateTimeStrings(catalogueItemIdData.created_time)}
-                </Typography>
+                <Grid item xs={12}>
```
Comment (joelvdavies) [inline src/catalogue/items/CatalogueItemsDetailsPanel.component.tsx:175]:
```suggestion
```

File: src/items/ItemsDetailsPanel.component.tsx [hunk 5]
```diff
@@ -133,17 +133,21 @@ function ItemsDetailsPanel(props: ItemsDetailsPanelProps) {
               </Grid>
 
               <Grid item xs={12} sm={6} key={7}>
-                <Typography color="text.primary">Last Modified</Typography>
-                <Typography color="text.secondary">
-                  {formatDateTimeStrings(itemData.modified_time)}
-                </Typography>
+                <Grid item xs={12}>
```
Comment (joelvdavies) [inline src/items/ItemsDetailsPanel.component.tsx:136]:
```suggestion
```

File: src/items/ItemsDetailsPanel.component.tsx [hunk 6]
```diff
@@ -133,17 +133,21 @@ function ItemsDetailsPanel(props: ItemsDetailsPanelProps) {
...
+                  </Typography>
+                </Grid>
               </Grid>
 
               <Grid item xs={12} sm={6} key={8}>
-                <Typography color="text.primary">Created</Typography>
-                <Typography color="text.secondary">
-                  {formatDateTimeStrings(itemData.created_time)}
-                </Typography>
+                <Grid item xs={12}>
```
Comment (joelvdavies) [inline src/items/ItemsDetailsPanel.component.tsx:145]:
```suggestion
```

File: src/systems/systemsTableView.component.tsx [hunk 7]
```diff
@@ -35,11 +36,12 @@ export const SystemsTableView = (props: SystemsTableViewProps) => {
   );
 
   const noResultsText = 'No systems found';
-  const columns = React.useMemo<MRT_ColumnDef<System>[]>(
-    () => [
+  const columns = React.useMemo<MRT_ColumnDef<System>[]>(() => {
+    return [
```
Comment (joelvdavies) [inline src/systems/systemsTableView.component.tsx:40]:
```suggestion
const columns = React.useMemo<MRT_ColumnDef<System>[]>(
    () => [
```

File: src/utils.tsx [hunk 8]
```diff
@@ -76,8 +76,14 @@ export const trimStringValues = (object: any): any => {
   return object;
 };
 
-export const formatDateTimeStrings = (dateTime: string): string => {
+export const formatDateTimeStrings = (
+  dateTime: string,
+  includesTime: boolean
```
Comment (joelvdavies) [inline src/utils.tsx:81]:
```suggestion
  includeTime: boolean
```

File: src/utils.tsx [hunk 9]
```diff
@@ -76,8 +76,14 @@ export const trimStringValues = (object: any): any => {
...
+): string => {
   const date = parseISO(dateTime);
-  const formattedDate = format(date, 'dd MMM yyyy HH:mm');
+  const formattedDate = includesTime
```
Comment (joelvdavies) [inline src/utils.tsx:84]:
```suggestion
  const formattedDate = includeTime
```

**Review by joelvdavies — COMMENTED** [review]
Minor code related things, LGTM though.

File: src/systems/systemsTableView.component.tsx (hunk 7 above)
Comment (MatteoGuarnaccia5) [reply src/systems/systemsTableView.component.tsx:40]: This just errors out, I think the return is needed as it is multiple objects (columns) now. Like in all the other table files

Comment (joelvdavies) [reply src/systems/systemsTableView.component.tsx:40]: It works if I use
```
  const columns = React.useMemo<MRT_ColumnDef<System>[]>(
    () => [
      {
        header: 'Name',
        accessorKey: 'name',
        size: 400,
        Cell: ({ renderedCellValue, row }) => {
          const canPlaceHere =
            type === 'copyTo' || !selectedSystemIds.includes(row.original.id);
          return (
            <Typography
              sx={{
                color: canPlaceHere ? 'inherit' : 'action.disabled',
              }}
            >
              {renderedCellValue}
            </Typography>
          );
        },
      },
      {
        header: 'Last modified',
        accessorFn: (row) => new Date(row.modified_time),
        id: 'modified_time',
        filterVariant: 'datetime-range',
        size: 400,
        enableGrouping: false,
        Cell: ({ row }) =>
          row.original.modified_time &&
          formatDateTimeStrings(row.original.modified_time, true),
      },
    ],
    [selectedSystemIds, type]
  );
```

Comment (MatteoGuarnaccia5) [reply src/systems/systemsTableView.component.tsx:40]: ahhh, I'd forgotten to change the semicolon after the closing bracket to a comma

File: src/manufacturer/manufacturer.component.tsx (hunk 2 above)
Comment (joelvdavies) [reply src/manufacturer/manufacturer.component.tsx:75]: @MatteoGuarnaccia5 I think you may have missed this one

**Review by joshuadkitenge — APPROVED** [review]

**Review by joelvdavies — DISMISSED** [review]

**Review by joshuadkitenge — APPROVED** [review]

**Review by joelvdavies — APPROVED** [review]

**Pull request merged**